src/PSRule.Rules.Azure/rules/Azure.RSV.Rule.ps1

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#
+# Validation rules for Azure Recovery Services Vault (RSV)
+#
+
+# Synopsis: Recovery Services Vault (RSV) not using geo-replicated storage (GRS) may be at risk.
+Rule 'Azure.RSV.StorageType' -Type 'Microsoft.RecoveryServices/vaults', 'Microsoft.RecoveryServices/vaults/backupconfig' -Tag @{ release = 'GA'; ruleSet = '2022_03' } {
+    $backupConfig = @($TargetObject);
+    if ($PSRule.TargetType -eq 'Microsoft.RecoveryServices/vaults') {
+        $backupConfig = @(GetSubResources -ResourceType 'Microsoft.RecoveryServices/vaults/backupconfig');
+    }
+
+    if ($backupConfig.Length -eq 0) {
+        return $Assert.Pass();
+    }
+
+    if($backupConfig.Properties.storageType){
+        write-host $backupConfig.name
+        write-host "backupConfig.Properties.storageType Value:"
+        write-host $backupConfig.Properties.storageType
+        write-host "TargetObject.Properties.storageType Value:"
+        write-host $TargetObject.Properties.storageType
+        $Assert.In($backupConfig, 'Properties.storageType', @(
+            'ReadAccessGeoZoneRedundant',
+            'GeoRedundant'
+        ));
+    } else {
+        return $Assert.Pass();
+    }
```

Comment:
@ms-sambell Honestly you mostly there. I think it is to do with the array on line 10. Try something like this instead.

```suggestion
    foreach ($config in $backupConfigs) {
        $Assert.AnyOf(
            $Assert.NotHasField($config, 'Properties.storageType'),
            $Assert.In($config, 'Properties.storageType', @(
                'ReadAccessGeoZoneRedundant',
                'GeoRedundant'
            ))
        )
    }
```

Reply:
Arrh! That worked. 

I didn't think I needed to do the foreach since the properties appear to map to the existing key/value in $TargetObject. 

Thank you for your help, this should be ready to go now.